mslib/utils/view_restoration.py

```diff
@@ -0,0 +1,162 @@
+
+    Licensed under the Apache License, Version 2.0 (the "License");
+    you may not use this file except in compliance with the License.
+    You may obtain a copy of the License at
+
+       http://www.apache.org/licenses/LICENSE-2.0
+
+    Unless required by applicable law or agreed to in writing, software
+    distributed under the License is distributed on an "AS IS" BASIS,
+    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+    See the License for the specific language governing permissions and
+    limitations under the License.
+"""
+
+import json
+import logging
+from PyQt5 import QtCore
+from pathlib import Path
+from PyQt5 import QtGui
+from mslib.msui import constants
+from mslib import __version__ as mss_version
+
+
+def save_view_settings(settings, global_data, flight_track_name):
+    """
+    Save view settings (for top, side, linear, and table views) to a JSON file.
+    """
+    if not isinstance(settings, list):
+        raise TypeError("Settings must be a list of dictionaries")
+    for setting in settings:
+        if not isinstance(setting, dict) or "view_type" not in setting:
+            raise ValueError("Each setting must be a dictionary with a 'view_type' key")
+    if not isinstance(global_data, dict):
+        logging.warning("Invalid global_data; using empty dictionary")
+        global_data = {}
+    if not isinstance(flight_track_name, str):
+        logging.warning("Invalid flight_track_name '%s'; converting to string", flight_track_name)
+        flight_track_name = str(flight_track_name)
+
+    try:
+        config_path = Path(constants.MSUI_CONFIG_PATH)
+        config_path.mkdir(parents=True, exist_ok=True)
+        settings_file = config_path / "view_settings.json"
+        settings_data = {}
+        if settings_file.exists():
+            try:
+                with settings_file.open("r", encoding="utf-8") as f:
+                    settings_data = json.load(f)
+            except json.JSONDecodeError:
+                logging.warning("Corrupted view_settings.json, initializing new file")
+
+        settings_data[flight_track_name] = {
+            "global": global_data,
+            "views": settings
+        }
+        with settings_file.open("w", encoding="utf-8") as f:
+            json.dump(settings_data, f, indent=2)
+        return True
+    except Exception as e:
+        logging.error("Failed to save view settings for %s: %s", flight_track_name, str(e))
+        return False
+
+
+def set_global_data(flight_track=None):
+    """
+    Create the global section for view settings using waypoints from flight_track or topview.
+    """
+    if flight_track:
+        flight_track_name = flight_track.name
+    else:
+        logging.warning("No flight track provided; using default flight track name")
+
+    return {
+        "mss_version": str(mss_version),
+        "flight_track_name": str(flight_track_name)
+    }
+
+
+def serialize_settings(settings_list):
```

Comment:
I think this code can be simplified quite a bit by using the `default` argument to json's dump functions instead: https://docs.python.org/3/library/json.html#:~:text=default%20(callable%20%7C%20None)%20%E2%80%93%20A%20function%20that%20is%20called%20for%20objects%20that%20can%E2%80%99t%20otherwise%20be%20serialized.%20It%20should%20return%20a%20JSON%20encodable%20version%20of%20the%20object%20or%20raise%20a%20TypeError.%20If%20None%20(the%20default)%2C%20TypeError%20is%20raised.